### PR TITLE
Support the Lenovo K6 Power

### PR DESCRIPTION
**Pull request by shwsh**
What works:

- Kernel compiles
- System/boot can be flashed
- USB Networking (SSH/Telnet)
- Weston loads
- Touchscreen
- Wifi
- Xwayland

### CODE REVIEW

**Comment by PureTryOut** [comment]
Awesome job!

Please create a device page on the wiki and add it to [the list of supported devices](https://wiki.postmarketos.org/wiki/Devices).

**Comment by shwsh** [comment]
@PureTryOut done and done!

File: aports/device/device-lenovo-karate/fb-workaround.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	do
+		su -c 'echo "0,0" > /sys/class/graphics/fb0/pan'
+	done
+}
+
+# tty1 autologin
+if [ $(tty) = "/dev/tty1" ]; then
+	# Run in background, to make /etc/profile not wait for it to finish
+	main &
+fi
```
Comment (ata2001) [inline aports/device/device-lenovo-karate/fb-workaround.sh]: Is this script necessary? If yes, please use `mdss-fb-init-hack` (https://github.com/postmarketOS/pmbootstrap/pull/927/files#diff-c0619f74670d5671f99810ec38e79dfbR8) instead.

File: aports/firmware/firmware-lenovo-karate/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,35 @@
+pkgname=firmware-lenovo-karate
+pkgver=1
+pkgrel=1
```
Comment (ata2001) [inline aports/firmware/firmware-lenovo-karate/APKBUILD]: When you create a new aport, please make sure, the pkgrel is 0.

File: aports/device/device-lenovo-karate/fb-workaround.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+main() {
```
Comment (drebrez) [inline aports/device/device-lenovo-karate/fb-workaround.sh]: @shwsh can you please try replacing this workaround script with the `mdss-fb-init-hack` package like it's been done [here](https://github.com/postmarketOS/pmbootstrap/commit/c79da69a1e13bba610a7bc6b6a00eca963ba03cc#diff-c0619f74670d5671f99810ec38e79dfbL8)?

File: aports/device/device-lenovo-karate/90-android-touch-dev.rules [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# use 'weston-calibrator' to calibrate the touchscreen!
+SUBSYSTEM=="input", ATTRS{name}=="fts_ts", \
+ENV{ID_INPUT}="1", ENV{ID_INPUT_TOUCHSCREEN}="1", \
+ENV{LIBINPUT_CALIBRATION_MATRIX}="0.975358 0.020484 -20.695250 0.031747 1.034592 -60.641296"
```
Comment (ollieparanoid) [inline aports/device/device-lenovo-karate/90-android-touch-dev.rules]: This line looks like the screen is already calibrated in the factory. Could you try to remove this line, and the `\` at the end of the line above, and try if it is still working?

If it works without extra calibration, please also remove the comment at the top `# use 'weston-calibrator' to calibrate the touchscreen!` (which makes it sound like it needs calibration).

File: aports/device/linux-lenovo-karate/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,81 @@
+# Kernel config based on: arch/arm64/configs/(CHANGEME!)
```
Comment (ollieparanoid) [inline aports/device/linux-lenovo-karate/APKBUILD]: Please adjust that line.

File: aports/device/device-lenovo-karate/90-android-touch-dev.rules (hunk 4 above)
Comment (shwsh) [reply aports/device/device-lenovo-karate/90-android-touch-dev.rules]: It doesn't work without the calibration, what should I do?

Comment (ollieparanoid) [reply aports/device/device-lenovo-karate/90-android-touch-dev.rules]: Then leave it as it is 👍

**Pull request merged**